src/odemis/gui/comp/overlay/polygon.py

```diff
@@ -71,6 +75,25 @@ def move_to(self, pos):
         self._phys_to_view()
         self.points.value = self._points
 
+    def get_state(self):
+        """Get the current state of the shape."""
+        state = {
+            PolygonStateKeys.V_POINTS: self.v_points.copy(),
+            PolygonStateKeys.FINISHED_PRIVATE: self._finished,
+            PolygonStateKeys.POINTS_PRIVATE: self._points.copy(),
+        }
+        # Only return the state if the polygon creation is finished
+        # By doing so avoid storing an undo action during polygon creation
+        return state if self.right_click_finished else {}
+
+    def restore_state(self, state: Dict[PolygonStateKeys, Any]):
+        """Restore the shape to a given state."""
+        self.v_points = state[PolygonStateKeys.V_POINTS]
```

Comment:
`v_points` are just the `_points` in the current view coordinate. They are going to be recomputed based on _points in the call to `_phys_to_view()`. So what's the point to store them? I'd suggest to not store them, and just reset v_points to `[None] * len(_points)` to make `_phys_to_view()` happy.